app/javascript/v3/helpers/BrowserHelper.js

```diff
@@ -0,0 +1,13 @@
+export const getBrowserLocale = enabledLanguages => {
+  const localeWithVariant = window.navigator.language.replace('-', '_');
+  const localeWithoutVariant = localeWithVariant.split('_')[0];
+  return enabledLanguages.find(
+    lang =>
+      lang.iso_639_1_code === localeWithVariant ||
+      lang.iso_639_1_code === localeWithoutVariant
+  )?.iso_639_1_code;
+};
```

Comment:
Instead of using `.replace`, we can use destructuring to split the language code into its language and region, then combine them to get the full locale. And, we can destructure the lang during the search process.

```suggestion
export const getBrowserLocale = enabledLanguages => {
  const [language, region] = navigator.language.split('-');
  const fullLocale = `${language}_${region || ''}`;
  return enabledLanguages.find(
    ({ iso_639_1_code }) =>
      iso_639_1_code === fullLocale || iso_639_1_code === language
  )?.iso_639_1_code;
};
```